src/io/flutter/inspector/DiagnosticsNode.java

```diff
@@ -612,29 +622,73 @@ public InspectorInstanceRef getDartDiagnosticRef() {
       if (location != null) {
         final ArrayList<Location> parameterLocations = location.getParameterLocations();
         if (parameterLocations != null) {
-            final Map<String, Location> names = new HashMap<>();
-            for (Location pl : parameterLocations) {
-              final String name = pl.getName();
-              if (name != null) {
-                names.put(name, pl);
-              }
+          final Map<String, Location> names = new HashMap<>();
+          for (Location pl : parameterLocations) {
+            final String name = pl.getName();
+            if (name != null) {
+              names.put(name, pl);
             }
-            for (DiagnosticsNode node : nodes) {
-              node.setParent(this);
-              final String name = node.getName();
-              if (name != null) {
-                final Location parameterLocation = names.get(name);
-                if (parameterLocation != null) {
-                  node.setCreationLocation(parameterLocation);
-                }
+          }
+          for (DiagnosticsNode node : nodes) {
+            node.setParent(this);
+            final String name = node.getName();
+            if (name != null) {
+              final Location parameterLocation = names.get(name);
+              if (parameterLocation != null) {
+                node.setCreationLocation(parameterLocation);
               }
             }
+          }
         }
       }
       return nodes;
     });
   }
 
+  @NotNull
+  public CompletableFuture<String> getPropertyDoc() {
+    if (propertyDocFuture == null) {
+      propertyDocFuture = createPropertyDocFurure();
+    }
+    return propertyDocFuture;
+  }
+
+  private CompletableFuture<String> createPropertyDocFurure() {
+    final DiagnosticsNode parent = getParent();
+    if (parent != null) {
+      final InspectorService inspectorService = parent.getInspectorService();
```

Comment:
nit: just use the inspectorService field of this Diagnostic directly instead of calling
parent.getInspectorService()

Reply:
👍